sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/V1Table.scala

```diff
@@ -109,6 +109,45 @@ private[sql] object V1Table {
       case _ => None
     }
   }
+
+  def toCatalogTable(
+      catalog: CatalogPlugin,
+      ident: Identifier,
+      t: MetadataOnlyTable): CatalogTable = {
+    val tableType = t.getTableType() match {
+      case TableSummary.VIEW_TABLE_TYPE => CatalogTableType.VIEW
+      case TableSummary.MANAGED_TABLE_TYPE => CatalogTableType.MANAGED
+      case _ => CatalogTableType.EXTERNAL
+    }
+    val viewText = Option(t.getViewText())
+    val tableProps = t.getTableProps().asScala.toMap
+    val (partCols, bucketSpec, clusterBySpec) = t.partitioning().toSeq.convertTransforms
+    CatalogTable(
+      identifier = TableIdentifier(
+        table = ident.name(),
+        database = Some(ident.namespace().lastOption.getOrElse("root")),
+        catalog = Some(catalog.name())),
+      tableType = tableType,
+      storage = CatalogStorageFormat.empty.copy(
```

Comment:
curious, do we not need to set inputFormat/outputFormat?

Reply:
`V1Table` itself does not access the input/output format when translating v1 table to v2 table. I think Hive table is never supported by DS v2 and we will not handle it here.